rest_framework/relations.py

```diff
@@ -331,6 +334,13 @@ def get_url(self, obj, view_name, request, format):
         return self.reverse(view_name, kwargs=kwargs, request=request, format=format)
 
     def to_internal_value(self, data):
+
+        if isinstance(data, self.get_queryset().model):
+            try:
+                return self.get_queryset().get(**{self.lookup_url_kwarg: getattr(data, self.lookup_field)})
```

Comment:
Is there a reason to perform the query instead of just returning the `data` as is?

Reply:
Yes. In the first iteration of this change, `data` is returned as is. Truthfully, it is simpler. However, it would be possible to have an object that isn't in the queryset of the serializer. If that is the case, there should be an error to that effect as the object is invalid for the serializer, like for a primary key.

Reply:
Then this handling should apply to `PrimaryKeyRelatedField` too, right?

Reply:
It does. It only keeps the primary key and then passes it to the existing `try`/`except` for the primary key field.